### ISSUE TITLE
How can I reproduce PPL=25 for Billion Word Benchmark

### ISSUE THREAD

**Issue by lanwuwei** [pytorch/fairseq#822]
I tried the following configurations to reproduce results from Transformer with tied adaptive embeddings (Baevski and Auli, 2018):
`Namespace(adaptive_input=True, adaptive_input_cutoff='60000,100000,640000', adaptive_input_factor=4, adaptive_softmax_cutoff='60000,100000,640000', adaptive_softmax_dropout=0, adaptive_softmax_factor=4, add_bos_token=False, arch='transformer_lm_gbw', atom='word', attention_dropout=0.1, block_lr=0.99, block_momentum=0.01, bucket_cap_mb=25, char_embedder_highway_layers=2, character_embedding_dim=4, character_embeddings=False, character_filters='[(1, 64), (2, 128), (3, 192), (4, 256), (5, 256), (6, 256), (7, 256)]', clip_norm=0.1, cpu=False, criterion='adaptive_loss', curriculum=0, data='data-raw/billion/', ddp_backend='no_c10d', debug=False, decoder_attention_heads=16, decoder_embed_dim=1024, decoder_ffn_embed_dim=4096, decoder_input_dim=1024, decoder_layers=16, decoder_learned_pos=False, decoder_normalize_before=True, decoder_output_dim=1024, device_id=3, distributed_backend='nccl', distributed_init_method='tcp://10.12.192.20:2992', distributed_port=-1, distributed_rank=3, distributed_world_size=12, dropout=0.1, fix_batches_to_gpus=False, fp16=False, fp16_init_scale=128, fp16_scale_tolerance=0.0, fp16_scale_window=None, future_target=False, keep_interval_updates=-1, keep_last_epochs=-1, lazy_load=False, load_extra=False, loader='utf-8', log_format=None, log_interval=1000, lr=[0.25], lr_period_updates=5000, lr_scheduler='cosine', lr_shrink=0.75, max_epoch=50, max_lr=1.0, max_sentences=None, max_sentences_valid=None, max_tokens=2048, max_update=0, memory_efficient_fp16=False, min_loss_scale=0.0001, min_lr=1e-09, momentum=0.99, no_decoder_final_norm=True, no_epoch_checkpoints=False, no_progress_bar=False, no_save=False, no_token_positional_embeddings=False, num_split=10, num_workers=1, optimizer='nag', optimizer_overrides='{}', output_dictionary_size=-1, parallel_model=False, past_target=False, queue_size=8, raw_text=True, relu_dropout=0.0, reset_lr_scheduler=False, reset_optimizer=False, restore_file='checkpoint_last.pt', sample_break_mode='eos', save_dir='checkpoints/trans_gbw', save_interval=1, save_interval_updates=1000, seed=1, self_target=False, sentence_avg=False, share_decoder_input_output_embed=False, skip_invalid_size_inputs_valid_test=False, sync_for_lr=500, sync_scheduler='4,10,50', t_mult=1, task='language_modeling', tensorboard_logdir='', threshold_loss_scale=None, tie_adaptive_proj=False, tie_adaptive_weights=True, tmpdir='tmptransform', tokens_per_sample=1024, train_subset='train', trainlist='billionlist', update_freq=[1], user_dir=None, valid_subset='valid', validate_interval=0, warmup_init_lr=0.001, warmup_updates=16000, weight_decay=5e-06)`

But the result I got is just below 40, which is far from PPL=25 in original paper. Can anyone help me point out the problem about my configuration?

**Comment by pipibjc** [comment]
@alexeib, would you mind to take a look? Thanks!

**Comment by yilegu** [comment]
I have the same question, as I also wanted to know how to reproduce Billion Word benchmark for paper by Baevski and Auli, 2018. Can anybody help out?

**Comment by apeterswu** [comment]
Same question raised. Could anyone give a setting for gbw data training?

**Comment by lematt1991** [comment]
Closing in favor of #898
